subprojects/language-native/src/main/java/org/gradle/language/nativeplatform/internal/incremental/DefaultSourceIncludesResolver.java

```diff
@@ -356,11 +269,12 @@ public int hashCode() {
         void visitUnresolved();
     }
 
-    private static class BuildableResult implements IncludeResolutionResult {
+    private static class BuildableResult implements IncludeResolutionResult, DefaultSourceIncludesSearchPath.SearchResult {
```

Comment:
I think if we just return the SearchResult, we don't have to implement this here.